src/libmongoc/tests/test-libmongoc.c

```diff
@@ -3023,5 +3023,6 @@ test_framework_skip_if_no_getlasterror (void) {
 
 bool
 test_framework_is_loadbalanced (void) {
-   return test_framework_getenv_bool ("MONGOC_TEST_LOADBALANCED");
+   return test_framework_getenv_bool ("MONGOC_TEST_LOADBALANCED") ||
+          test_framework_getenv_bool ("MONGOC_TEST_DNS_LOADBALANCED");
```

Comment:
This check enables mocking the `serviceID` to successfully connect to servers when `loadBalanced=true` is specified in the URI. The rationale is described in the Load Balancer specification's [test README](https://github.com/mongodb/specifications/tree/2b0e9ed07446b2a92ac66216d19da59b7f582e69/source/load-balancers/tests#test-runner-configuration).

`MONGOC_TEST_DNS_LOADBALANCED` only applies to DNS tests. It was added for consistency with existing DNS tests. It is planned for removal in CDRIVER-4045.